sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -350,6 +389,23 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 	return events
 }
 
+// skipIntegrationCreate returns true if image integrations should NOT be created from this secret.
+func skipIntegrationCreate(secret *v1.Secret) bool {
+	if features.SourcedAutogeneratedIntegrations.Enabled() {
+		// Create integration if sourced autogen enabled.
+		return false
```

Comment:
so we still want to make the duplicate?

Reply:
Yes, when `SourcedAutogeneratedIntegrations` is enabled, all secrets are to be referenced by their namespace+name.  If, for example, a namespace has 10 duplicate secrets with different names, each of those secrets needs to be stored so that when a workflow references any one of the 10 secrets we can match / use it.

The global pull secret is an exception and can be used by any workload regardless, a duplicate just takes up space and competes with normal autogen integrations reducing the chances of a different secret 'winning' and making it into an image integration.